src/handlers/issue/generate-permits.ts

```diff
@@ -88,6 +88,27 @@ async function generateComment(totals: TotalsById, issue: GitHubIssue, config: B
       ...erc20Permits.map((permit) => ({ type: "erc20-permit", ...permit })),
       ...erc721Permits.map((nftMint) => ({ type: "erc721-permit", ...nftMint })),
     ];
+
+    const { error: insertError } = await supabase.from("permits").insert(
+      [
+        erc20Permits.map((claim) => ({
+          amount: claim.permit.permitted.amount,
+          nonce: claim.permit.nonce,
+          deadline: claim.permit.deadline,
+          signature: claim.signature,
+          beneficiary_id: parseInt(userId),
+        })),
+        erc721Permits.map((claim) => ({
+          amount: "1",
+          nonce: claim.request.nonce,
+          deadline: claim.request.deadline,
+          signature: claim.signature,
+          beneficiary_id: parseInt(userId),
+        })),
+      ].flat()
+    );
+    if (insertError) throw insertError;
```

Comment:
Actually I just realized something, but, this should all be wrapped in some type of try/catch. 

If writing to the database fails for any reason, the bot will not comment the rewards. I feel that this can make this system brittle and its mission critical to remain operational. 

What do you guys think?

Reply:
good point actually. you can see same logic on line 53 too while reading from wallets.

if not-inserting-permit is an error that can be ignored, I think comment first and then write to db approach looks better.

Reply:
in any case it should be a separate task I think which focuses on error handling rather than writing permits to db only.